src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -305,25 +315,31 @@ private static bool TryGetBoostedValue(ObjectInstance valueToCheck, out JsValue
             return true;
         }
 
-        private static JsValue TryDetectDynamicFieldCreation(string property, ObjectInstance valueAsObject, IndexField field)
+        private JsValue TryDetectDynamicFieldCreation(string property, ObjectInstance valueAsObject, ref IndexField field)
```

Comment:
Why do you use `ref`? I though the goal is to avoid modifying it completely

Reply:
We're not modifying underlying object inside `IndexDefinition` but value of local `field` variable.  I was thinking about it, but in current flow that would mean doing a lot `out`. I would prefer just working with `ref`.

Reply:
Not perfect but we can leave with that.